src/probability/kernel/condexp.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2023 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import probability.kernel.cond_distrib
+
+/-!
+# Kernel associated with a conditional expectation
+
+We define `condexp_kernel μ m`, a kernel from `Ω` to `Ω` such that for all integrable, strongly
+measurable `f`, `μ[f | m] =ᵐ[μ] λ ω, ∫ y, f y ∂(condexp_kernel μ m ω)`.
+
+## Main definitions
+
+* `condexp_kernel μ m`: kernel such that `μ[f | m] =ᵐ[μ] λ ω, ∫ y, f y ∂(condexp_kernel μ m ω)`.
+
+## Main statements
+
+* `condexp_ae_eq_integral_condexp_kernel`: `μ[f | m] =ᵐ[μ] λ ω, ∫ y, f y ∂(condexp_kernel μ m ω)`.
+
+-/
+
+open measure_theory set filter topological_space
+
+open_locale ennreal measure_theory probability_theory
+
+namespace probability_theory
+
+variables {Ω F : Type*} [topological_space Ω] {m : measurable_space Ω} [mΩ : measurable_space Ω]
+  [polish_space Ω] [borel_space Ω] [nonempty Ω]
+  {μ : measure Ω} [is_finite_measure μ]
+  [normed_add_comm_group F] [normed_space ℝ F] [complete_space F]
+
+/-- Kernel associated with the conditional expectation with respect to a σ-algebra. -/
+@[irreducible] noncomputable
+def condexp_kernel (μ : measure Ω) [is_finite_measure μ] (m : measurable_space Ω) :
+  @kernel Ω Ω m mΩ :=
+@cond_distrib Ω Ω Ω _ mΩ _ _ _ mΩ m id id μ _
+
+/-- The conditional expectation of `f` with respect to a σ-algebra `m` is almost everywhere equal to
+the integral `∫ y, f y ∂(condexp_kernel μ m ω)`. -/
+lemma condexp_ae_eq_integral_condexp_kernel (hm : m ≤ mΩ)
+  {f : Ω → F} (hf : strongly_measurable f) (hf_int : integrable f μ) :
```

Comment:
Do you think the assumption `hf` is really necessary, or is it an artifact of the current implementation? For instance, is it true that if `f` is ae measurable for mu, then for almost every omega `f` is also ae measurable for `condexp_kernel μ m ω`?

Reply:
I kept the PR marked WIP for two days because I wanted to investigate this precise question more, but I then forgot about it and switched to awaiting-review. Sorry about that. I have no intuition about ae-measurability with respect to condexp_kernel for now, but I'll see what I can get!

Reply:
My guess is that it's true: if `f` is just ae strongly measurable, it coincides almost everywhere (say on a set `s`) with a strongly measurable function `f'`. Then `s` has full measure wrt to the conditional measure `condexp_kernel mu m omega` for almost every omega, which means that for all these omega then f is ae strongly measurable (with witness function `f'`).

Reply:
I removed the strongly measurable assumption. I did not add any result about measurability or integrability with respect to `condexp_kernel` though. I changed the proof to use the lemma `condexp_prod_ae_eq_integral_cond_distrib'`, which does not need strongly_measurable but requires integrability with respect to a map of the measure, and I proved that integrability using that the map has a very particular form.

Reply:
I added several lemmas about `ae_strongly_measurable` and `integrable` with respect to `cond_distrib` and `condexp_kernel`.